libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -60,10 +60,9 @@ void MaterialBuilderBase::prepare() {
         mShaderModels.set(static_cast<size_t>(ShaderModel::GL_CORE_41));
     }
 
-    // If the code gen target API was specifically set to Vulkan, generate for Vulkan, otherwise
-    // generate for OpenGL (case OPENGL or ALL)
-    TargetApi glCodeGenTargetApi = mCodeGenTargetApi != TargetApi::VULKAN ?
-            TargetApi::OPENGL : TargetApi::VULKAN;
+    // OpenGL is a special case. If we're doing any optimization, then we need to go to Spir-V.
+    TargetApi glCodeGenTargetApi = mOptimization > MaterialBuilder::Optimization::PREPROCESSOR ?
```

Comment:
That is not quite correct here. If the target API is vulkan, the codegen API must always be Vulkan. So when the optimization level is <= PREPROCESSOR we must return either OPENGL or VULKAN.

Reply:
Correct, this variable is only for OpenGL.

A few lines down:
```
mCodeGenPermutations.push_back({i, TargetApi::VULKAN, TargetApi::VULKAN});
```

For Vulkan, the target API is always `TargetApi::VULKAN`.

Reply:
Yeah, this is important.  The GLSL that we generate for Vulkan is (unfortunately) slightly different from the GLSL that we generate for OpenGL.

Reply:
Sounds good then. I'm glad we're ditching this hack in the material builder API.